### PR TITLE
Safer reindex

### PR DESCRIPTION
**Pull request by czue**
pr feedback from https://github.com/dimagi/commcare-hq/pull/11161

@snopoke cc @nickpell

### CODE REVIEW

File: corehq/ex-submodules/pillowtop/reindexer/reindexer.py [hunk 1]
```diff
@@ -26,15 +25,15 @@ def __init__(self, pillow, change_provider, elasticsearch, index_info):
...
         super(ElasticPillowReindexer, self).reindex(start_from)
 
         self._prepare_index_for_usage()
 
-    def _delete_and_prepare_index_for_reindex(self):
-        ensure_production_index_deleted(self.index_info.index)
-        self.es.indices.create(index=self.index_info.index, body=self.index_info.meta)
+    def _prepare_index_for_reindex(self):
+        if not self.es.indices.exists(self.index_info.index):
+            self.es.indices.create(index=self.index_info.index, body=self.index_info.meta)
```
Comment (snopoke) [inline corehq/ex-submodules/pillowtop/reindexer/reindexer.py]: Should there be a warning if the index exists and `start_from` is None?

**Comment by snopoke** [comment]
minor feedback but :+1: on :white_check_mark:

File: corehq/ex-submodules/pillowtop/reindexer/reindexer.py (hunk 1 above)
Comment (czue) [inline corehq/ex-submodules/pillowtop/reindexer/reindexer.py]: Seems reasonable. What did you have in mind? Like a logger call?

Comment (snopoke) [inline corehq/ex-submodules/pillowtop/reindexer/reindexer.py]: I had thought of a user confirmation but I see this class is isolated from the management command. Maybe not worth the effort but you could use a `force` param and raise an exception if it's `False`.

**Comment by czue** [comment]
:white_check_mark:

**Pull request merged**